### PR TITLE
Add missing context for Ed25519VerificationKey2020

### PR DESCRIPTION
**Pull request by matthieubosquet**
In its current form, the example DID document that is supposed to represent an Ed25519VerificationKey2020 usable for various purposes contains the following statements:

```ttl
<did:example:123> <https://w3id.org/security#assertionMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#authenticationMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#capabilityDelegationMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#capabilityInvocationMethod> <did:example:123#key-0> .
```

The missing contexts add the statements describing the public key:

```ttl
# Public key pointer
<did:example:123> <https://w3id.org/security#publicKey> <did:example:123#key-0> .

# Valid usage (only existing statements currently)
<did:example:123> <https://w3id.org/security#assertionMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#authenticationMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#capabilityDelegationMethod> <did:example:123#key-0> .
<did:example:123> <https://w3id.org/security#capabilityInvocationMethod> <did:example:123#key-0> .

# Public key description
<did:example:123#key-0> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://w3id.org/security#Ed25519VerificationKey2020> .
<did:example:123#key-0> <https://w3id.org/security#controller> <did:example:123> .
<did:example:123#key-0> <https://w3id.org/security#publicKeyMultibase> "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"^^<https://w3id.org/security#multibase> .
```

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -360,10 +360,12 @@ <h3>Ed25519VerificationKey2020</h3>
             }
           </pre>
 
-          <pre class="example" title="Example in DID Document.">
+          <pre class="example" title="Ed25519VerificationKey2020 in a DID Document.">
             {
               "@context": [
                 "https://www.w3.org/ns/did/v1",
+                "https://w3id.org/security/v1",
```
Comment (OR13) [inline index.html]:
```suggestion
```

Comment (matthieubosquet) [reply index.html]: I don't know if it's a problem with the [JSON-LD playground](https://json-ld.org/playground/), but without `https://w3id.org/security/v1` in context [I only get the four statements](https://tinyurl.com/2hkjvect) that I get without my PR.

Comment (OR13) [reply index.html]: Thats probably a problem with the 2020 context then... you should not need more than 2.

**Review by OR13 — CHANGES_REQUESTED** [review]
Pending close, this is my last action as editor.

**Comment by matthieubosquet** [comment]
@OR13 as mentionned in [my previous comment](https://github.com/w3c-ccg/di-eddsa-2020/pull/15#discussion_r983980896), the reason why I am not applying your suggestion is that:

- Without `https://w3id.org/security/suites/ed25519-2020/v1`, the following statement is missing:
   ```ttl
   <did:example:123#key-0> <https://w3id.org/security#publicKeyMultibase> "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"^^<https://w3id.org/security#multibase> .
   ```
   And the key type becomes `did:Ed25519VerificationKey2020` instead of `https://w3id.org/security#Ed25519VerificationKey2020` (which is not a correct IRI).
- Without `https://w3id.org/security/v1`, all statements about the public key are missing:
   ```ttl
   <did:example:123> <https://w3id.org/security#publicKey> <did:example:123#key-0> .
   <did:example:123#key-0> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://w3id.org/security#Ed25519VerificationKey2020> .
   <did:example:123#key-0> <https://w3id.org/security#controller> <did:example:123> .
   <did:example:123#key-0> <https://w3id.org/security#publicKeyMultibase> "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"^^<https://w3id.org/security#multibase> .
   ```
   
There is no technical issue with having more than two contexts, it is however problematic to have missing statements for RDF processors.

**Comment by OR13** [comment]
I'm no longer an editor for this item, I suggest you ping @dlongley and @msporny regarding your example.

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Yes, you need to remove "https://w3id.org/security/v1"... you don't need it here.

Comment (matthieubosquet) [reply index.html]: But @msporny, without `https://w3id.org/security/v1`, as per [my comment below](https://github.com/w3c-ccg/di-eddsa-2020/pull/15#issuecomment-1325680415), all the following statements about the public key are missing:

```ttl
<did:example:123> <https://w3id.org/security#publicKey> <did:example:123#key-0> .
<did:example:123#key-0> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://w3id.org/security#Ed25519VerificationKey2020> .
<did:example:123#key-0> <https://w3id.org/security#controller> <did:example:123> .
<did:example:123#key-0> <https://w3id.org/security#publicKeyMultibase> "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"^^<https://w3id.org/security#multibase> .
```

Therefore what do you mean by "the `https://w3id.org/security/v1` context is not needed?

Comment (dlongley) [reply index.html]: The problem with this example is that it uses `publicKey` when it should instead use `verificationMethod`. That should remove the need for the `security/v1` context and make it conformant with the DID spec (`publicKey` was used in very early versions of the DID spec and it was later changed to `verificationMethod`).

**Comment by matthieubosquet** [comment]
Followed up by removing the unnecessary context and [using the correct DID document property](https://github.com/w3c-ccg/di-eddsa-2020/pull/15/commits/f7accd2f1c5ce8ba6224194990c821e71ad3f7e3): `verificationMethod` instead of `publicKey` as per @dlongley's [suggestion](https://github.com/w3c-ccg/di-eddsa-2020/pull/15#discussion_r1038798102).

**Review by dlongley — APPROVED** [review]

**Comment by msporny** [comment]
Hey @matthieubosquet can you please fix the merge conflicts so we can merge this PR?

**Review by msporny — APPROVED** [review]

File: index.html [hunk 2]
```diff
@@ -385,19 +385,21 @@ <h3>Ed25519VerificationKey2020</h3>
...
-    "publicKeyMultibase": "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"
-  }],
+  "verificationMethod": [
+    {
+      "id": "did:example:123#key-0",
+      "type": "Ed25519VerificationKey2020",
+      "controller": "did:example:123",
+      "publicKeyMultibase": "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"
+    }
+  ],
```
Comment (matthieubosquet) [inline index.html]:
```suggestion
  "verificationMethod": [{
    "id": "did:example:123#key-0",
    "type": "Ed25519VerificationKey2020",
    "controller": "did:example:123",
    "publicKeyMultibase": "z6Mkf5rGMoatrSj1f4CyvuHBeXJELe9RPdzo2PKGNCKVtZxP"
  }],
```

**Comment by matthieubosquet** [comment]
@msporny after formatting, it looks like [the one significant change in this PR](https://github.com/w3c-ccg/di-eddsa-2020/pull/15/commits/f7accd2f1c5ce8ba6224194990c821e71ad3f7e3) ("publicKey": -> "verificationMethod") has been addressed in parallel elsewhere.

The only slightly inconsistent bit is an extra two spaces but we agree on the outcome. Feel free to merge or close.

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Editorial, multiple reviews, discussion, changes applied via another PR, this PR remains as a formatting fix, merging.

**Pull request merged**